image/image.go

```diff
@@ -122,6 +148,31 @@ type genericManifest interface {
 	GetLayers() []string
 }
 
+type descriptor struct {
+	MediaType string `json:"mediaType"`
+	Size      int64  `json:"size"`
+	Digest    string `json:"digest"`
+}
+
+type manifestSchema2 struct {
+	Config descriptor   `json:"config"`
+	Layers []descriptor `json:"layers"`
+}
+
+func (m *manifestSchema2) String() string {
+	// TODO: does it make sense to have string here?
```

Comment:
I think also we should try not to break what @giuseppe did for system containers.

Now the maybe-issues are::

- the manfiest can be v2s2 (till now it was only v2s1)
- in the layers you also get the _config_ layer (till now the config was embedded in v2s1)

@giuseppe, could you tell me if system containers can be adapted to this? or it's time to use `skopeo copy` in atomic